public_html/app/libraries/Core.php

```diff
@@ -12,10 +12,11 @@ class Core {
 
         public function __construct() {
             $url = $this->getUrl();
+            $url1 = array_shift($url);
             // Look in controllers for first value of URL
             if (file_exists(ROOT_DIR . '/app/controllers/' . ucwords($url[0]). '.php')) {
```

Comment:
move `/app/controllers/` to constant

Reply:
`private const CONTROLLER_DIR = '/app/controllers/';`

Reply:
If i do this, routing is not working and i can not acces different pages.
Fore example, http://localhost:8081/users/login or http://localhost:8081/users/register still redirect me to a homepage.

Reply:
Am, I think you don't understand me correctly. You should move **/app/controllers/** string to class constant.

`private const CONTROLLER_DIR` = '/app/constrollers/';

`if(file_exists(ROOT_DIR . self::CONTROLLER_DIR . ucwords($url[0]))`